balancer-design.typ

```diff
@@ -17,3 +17,44 @@ In Rust, packages are called "crates". The Balancer and Harness is split into mu
 	image("figures/balancer-crates.svg"),
 	caption: "Package diagram showing the Balancer and Harness crates.",
 ) <Figure::balancer-crates>
+
+== Dependencies
+
+#let cargo = toml("Cargo.toml")
+#let crates = cargo.workspace.dependencies
+
+#let build_table(data) = {
+  let keys = data.keys()
+	let rows = ()
+
+	for (key, value) in data {
+		if(type(value) == str) {
+			rows.push((key, value, ""))
+		} 
+		else if(type(value) == dictionary and value.keys().contains(version)) {
+			let row = (key, value.version,)
+			let features = ""
+			
+			for entry in value.features {
+				features += entry + ", "
+			}
+
+			row.push(features)
+			rows.push(row)
+		}
+		else {
+			rows.push((key, "", ""))
```

Comment:
This doesn't display the git repo/commit in the version column, nor any features that are enabled for those deps.